### PR TITLE
Fix 636

### PR DESCRIPTION
**Pull request by PaulJonasJost**
Fixes Issue #636.

I did not include a try...except for each case as the sampling is special in the sense that the Class of `result.sampling_result` changes from `SampleResult` to `McmcPtResult` when sampling is performed. This created the error.

### CODE REVIEW

File: pypesto/store/read_from_hdf5.py [hunk 1]
```diff
@@ -207,7 +207,11 @@ def read(self) -> Result:
             for key in f['/sampling/results'].attrs:
                 sample_result[key] = \
                     f['/sampling/results'].attrs[key]
-        self.results.sample_result = McmcPtResult(**sample_result)
+        try:
+            self.results.sample_result = McmcPtResult(**sample_result)
+        except TypeError:
+            logger.warning("Warning: You tried loading a non existent "
```
Comment (dilpath) [inline pypesto/store/read_from_hdf5.py:213]:
```suggestion
            logger.warning("Warning: You tried loading a non-existent "
```

File: pypesto/store/save_to_hdf5.py [hunk 2]
```diff
@@ -182,6 +185,11 @@ def write(self, result: Result, overwrite: bool = False):
         """
         Write HDF5 sampling file from pyPESTO result object.
         """
+        # if there is no sample available, raise a warning and return
```
Comment (dilpath) [inline pypesto/store/save_to_hdf5.py:188]:
```suggestion
        # if there is no sample available, log a warning and return
```

File: pypesto/store/save_to_hdf5.py [hunk 3]
```diff
@@ -182,6 +185,11 @@ def write(self, result: Result, overwrite: bool = False):
...
+        if(isinstance(result.sample_result, SampleResult)):
```
Comment (dilpath) [inline pypesto/store/save_to_hdf5.py:189]: Maybe add a comment about `SampleResult` being some dummy type that indicates no sampling has already been done (actual sampling is indicated by `McmcPtResult`)

**Review by dilpath — APPROVED** [review]
Thanks!

File: pypesto/store/save_to_hdf5.py (hunk 3 above)
Comment (PaulJonasJost) [reply pypesto/store/save_to_hdf5.py:189]:
```suggestion
        # SampleResult is only a dummy class created by the Result class
        if(isinstance(result.sample_result, SampleResult)):
```

**Pull request merged**